### PR TITLE
OV-205: Add voices modal

### PR DESCRIPTION
**Pull request by Alblupynos**
Add voices modal, update script voice in store
image
image

### CODE REVIEW

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/components/voices-modal/components/voice-card.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    isChecked: boolean;
+    onClick: (voice: Voice) => void;
+};
+
+const VoiceCard: React.FC<Properties> = ({ voice, isChecked, onClick }) => {
+    const [isPlaying, setIsPlaying] = useState(false);
+
+    const handlePlayClick = useCallback((): void => {
+        setIsPlaying((previous) => !previous);
+    }, []);
```
Comment (stefano-lacorazza) [inline frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/components/voices-modal/components/voice-card.tsx:23]: You could add event.stopPropagation(); so when the play button is clicked the modal doesnt close automatically.

File: frontend/src/bundles/common/components/modal/modal.module.css
Comment (o-nedashkivska) [inline frontend/src/bundles/common/components/modal/modal.module.css:1]: styles.module.css

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/components/script.tsx [hunk 2]
```diff
@@ -88,32 +95,45 @@ const Script: React.FC<Properties> = ({ id, text, url }) => {
...
-            <HStack justify="end" w="full" gap={0}>
-                <Tooltip
-                    isDisabled={Boolean(url)}
-                    label="Click to update audio"
-                    placement="top"
-                    hasArrow
+            <HStack justify="space-between" w="full">
+                <Text
+                    onClick={handleChangeVoiceId}
+                    cursor={'pointer'}
```
Comment (o-nedashkivska) [inline frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/components/script.tsx:107]:
```suggestion
                    cursor='pointer'
```

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/components/voices-modal/components/voice-card.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    const handleCardClick = useCallback((): void => {
+        onClick(voice);
+    }, [onClick, voice]);
+    return (
+        <Card
+            variant={isChecked ? 'outline' : 'elevated'}
+            cursor={'pointer'}
```
Comment (o-nedashkivska) [inline frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/components/voices-modal/components/voice-card.tsx:30]:
```suggestion
            cursor='pointer'
```

File: frontend/src/bundles/studio/components/video-menu/components/mock/voices-mock.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { v4 as uuidv4 } from 'uuid';
+
+import { type Voice } from '~/bundles/studio/types/types.js';
+
+const mockVoices: Voice[] = Array.from({ length: 10 }, (_, index) => ({
```
Comment (o-nedashkivska) [inline frontend/src/bundles/studio/components/video-menu/components/mock/voices-mock.ts:5]: Please, add todo comment to remove it

File: frontend/src/bundles/studio/store/slice.ts [hunk 5]
```diff
@@ -121,6 +129,12 @@ const { reducer, actions, name } = createSlice({
                 items: state.scripts,
             });
         },
+        changeScriptVoice(state, action: PayloadAction<ScriptVoicePayload>) {
+            const { scriptId, voice } = action.payload;
+            state.scripts = state.scripts.map((script) =>
+                script.id === scriptId ? { ...script, voice } : script,
+            );
+        },
```
Comment (o-nedashkivska) [inline frontend/src/bundles/studio/store/slice.ts:137]: Please, reuse editScript reducer for changing voice. And we are going to use `voiceName` property https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/236

File: frontend/src/bundles/studio/types/voice.type.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+type Voice = {
+    id: string;
+    name: string;
+    previewUrl?: string;
+    code: string;
+};
```
Comment (o-nedashkivska) [inline frontend/src/bundles/studio/types/voice.type.ts:6]: We already have voice type and it is not the same as you created. Make sure to use only those properties that exist in endpoint response.
Please review this pr https://github.com/BinaryStudioAcademy/bsa-2024-outreachvids/pull/185 to reuse Voice type and review what properties are returned.

File: frontend/src/bundles/common/components/components.ts [hunk 7]
```diff
@@ -16,8 +17,12 @@ export { Textarea } from './textarea/textarea.js';
 export { UploadVideo } from './upload-video/upload-video.js';
 export { VideoModal } from './video-modal/video-modal.js';
 export {
+    type ModalProps,
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/common/components/components.ts:20]: Import types from common/types

File: frontend/src/bundles/common/components/modal/modal.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { type ModalProps } from '@chakra-ui/react';
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/common/components/modal/modal.tsx:1]: Import from common export

File: frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/components/voices-modal/components/voices-modal-content.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
+import { Heading, SimpleGrid, VStack } from '@chakra-ui/react';
+
```
Comment (sofiia-trokhymchuk) [inline frontend/src/bundles/studio/components/video-menu/components/menu-content/script-content/components/voices-modal/components/voices-modal-content.tsx:2]: Import from common components

**Review by sofiia-trokhymchuk — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/studio/types/script.type.ts [hunk 10]
```diff
@@ -1,14 +1,15 @@
...
+import { type Voice } from './types.js';
 
 type Script = {
     id: string;
     duration: number;
     text: string;
-    voiceName: string;
     iconName: ValueOf<typeof PlayIconNames>;
     url?: string;
+    voice?: Voice | undefined;
```
Comment (o-nedashkivska) [inline frontend/src/bundles/studio/types/script.type.ts:12]:
```suggestion
    voice?: Voice;
```

Comment (Alblupynos) [reply frontend/src/bundles/studio/types/script.type.ts:12]: don't work without, dm you with an explanation

**Review by o-nedashkivska — APPROVED** [review]

**Review by sofiia-trokhymchuk — APPROVED** [review]

**Review by anton-otroshchenko — APPROVED** [review]

**Pull request merged**